### PR TITLE
Get homeworkby

### PR DESCRIPTION
**Pull request by betulEZ**
add Lesson and Delete, getHomework
add Test

### CODE REVIEW

File: backend/src/main/java/com/example/backend/controller/StudentController.java [hunk 1]
```diff
@@ -33,8 +34,16 @@ public Student getStudentById(@PathVariable String id){
     public Student update(@PathVariable String id,@RequestBody StudentDTO studentDTO) {
         return studentService.update(id, studentDTO);
     }
+    @PutMapping("/{id}/addLesson")
```
Comment (MarcellNF) [inline backend/src/main/java/com/example/backend/controller/StudentController.java:37]:
```suggestion
    @PutMapping("/{id}/add-lesson")
```

File: frontend/src/studentPage/StudentNavbar.tsx [hunk 2]
```diff
@@ -1,28 +1,106 @@
...
+        if (newLesson) {
+            setFormData({
+                ...newLesson,
+                id: newLesson.id,
+            });
+        }
+    }
+    const handleUpdate = async () => {
+        try {
+            const response = await axios.put(`/api/students/${props.studentId}/addLesson`, formData);
```
Comment (MarcellNF) [inline frontend/src/studentPage/StudentNavbar.tsx:65]:
```suggestion
            const response = await axios.put(`/api/students/${props.studentId}/add-lesson`, formData);
```

**Review by MarcellNF — CHANGES_REQUESTED** [review]
so far so good, few little things and you need to convert your draft to pr first

File: frontend/src/studentPage/StudentHomePage.tsx [hunk 3]
```diff
@@ -2,43 +2,99 @@ import StudentNavbar from "./StudentNavbar.tsx";
...
-                                <Button>Detail</Button>
-                            </CardActions>
+                            <div className="card-container">
+                                <div>
+                                    <DeleteIcon style={{color: 'red'}} onClick={() => handleLessonDelete(lesson)}/>
+                                </div>
+                                <div>
+                                    <Link to={`/homeworks/${lesson.id}`}>
+                                        <Badge badgeContent={homeworkCounts[lesson?.id] || 0} color="primary">
+                                            <AddHomeWorkIcon color="action"/>
```
Comment (jessinowski) [inline frontend/src/studentPage/StudentHomePage.tsx:86]: maybe you can try to put the styling in extra css

**Review by jessinowski — COMMENTED** [review]
LGTM, only a few comments. And you should try to fix your issues :)

File: backend/src/main/java/com/example/backend/controller/StudentController.java (hunk 1 above)
Comment (betulEZ) [reply backend/src/main/java/com/example/backend/controller/StudentController.java:37]: How can I my draft to pr convert?

**Review by MarcellNF — APPROVED** [review]
LGTM!:)

**Pull request merged**